File: testdata/syntax/sample.js
// Sample JavaScript file for syntax highlighting

const fibonacci = (n) => {
  if (n <= 1) return n;
  let a = 0, b = 1;
  for (let i = 2; i <= n; i++) {
    [a, b] = [b, a + b];
  }
  return b;
};

class Calculator {
  constructor() {
    this.history = [];
  }

  add(x, y) {
    const result = x + y;
    this.history.push(result);
    return result;
  }
}

const calc = new Calculator();
console.log(calc.add(1, 2));
console.log(fibonacci(10));

File: testdata/syntax/sample.py
#!/usr/bin/env python3
"""Sample Python file for syntax highlighting."""

def fibonacci(n: int) -> int:
    """Return the nth Fibonacci number."""
    if n <= 1:
        return n
    a, b = 0, 1
    for _ in range(2, n + 1):
        a, b = b, a + b
    return b

class Calculator:
    def __init__(self):
        self.history = []

    def add(self, x, y):
        result = x + y
        self.history.append(result)
        return result

if __name__ == "__main__":
    calc = Calculator()
    print(calc.add(1, 2))
    print(fibonacci(10))
